app/data_client.go

```diff
@@ -0,0 +1,781 @@
+		return nil, 0, "", err
+	}
+	dataArray := []TabularData{}
+	var metadata *pb.CaptureMetadata
+	for _, data := range resp.Data {
+		if len(resp.Metadata) != 0 && int(data.MetadataIndex) >= len(resp.Metadata) {
+			metadata = &pb.CaptureMetadata{}
+		} else {
+			metadata = resp.Metadata[data.MetadataIndex]
+		}
+		dataArray = append(dataArray, tabularDataFromProto(data, metadata))
+	}
+	return dataArray, resp.Count, resp.Last, nil
+}
+
+// TabularDataBySQL queries tabular data with a SQL query.
+func (d *Client) TabularDataBySQL(ctx context.Context, organizationID, sqlQuery string) ([]map[string]interface{}, error) {
+	resp, err := d.client.TabularDataBySQL(ctx, &pb.TabularDataBySQLRequest{
+		OrganizationId: organizationID,
+		SqlQuery:       sqlQuery,
+	})
+	if err != nil {
+		return nil, err
+	}
+	dataObjects, err := BsonToGo(resp.RawData)
+	if err != nil {
+		return nil, err
+	}
+	return dataObjects, nil
+}
+
+// TabularDataByMQL queries tabular data with an MQL (MongoDB Query Language) query.
+func (d *Client) TabularDataByMQL(ctx context.Context, organizationID string, mqlbinary [][]byte) ([]map[string]interface{}, error) {
+	resp, err := d.client.TabularDataByMQL(ctx, &pb.TabularDataByMQLRequest{
+		OrganizationId: organizationID,
+		MqlBinary:      mqlbinary,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := BsonToGo(resp.RawData)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// BinaryDataByFilter queries binary data and metadata based on given filters.
+func (d *Client) BinaryDataByFilter(
+	ctx context.Context,
+	filter Filter,
+	limit uint64,
+	sortOrder Order,
+	last string,
+	includeBinary bool,
+	countOnly bool,
+	includeInternalData bool,
+) ([]BinaryData, uint64, string, error) {
+	resp, err := d.client.BinaryDataByFilter(ctx, &pb.BinaryDataByFilterRequest{
+		DataRequest: &pb.DataRequest{
+			Filter:    filterToProto(filter),
+			Limit:     limit,
+			Last:      last,
+			SortOrder: orderToProto(sortOrder),
+		},
+		IncludeBinary:       includeBinary,
+		CountOnly:           countOnly,
+		IncludeInternalData: includeInternalData,
+	})
+	if err != nil {
+		return nil, 0, "", err
+	}
+	data := make([]BinaryData, len(resp.Data))
+	for i, protoData := range resp.Data {
+		data[i] = binaryDataFromProto(protoData)
+	}
+	return data, resp.Count, resp.Last, nil
+}
+
+// BinaryDataByIDs queries binary data and metadata based on given IDs.
+func (d *Client) BinaryDataByIDs(ctx context.Context, binaryIDs []BinaryID) ([]BinaryData, error) {
+	resp, err := d.client.BinaryDataByIDs(ctx, &pb.BinaryDataByIDsRequest{
+		IncludeBinary: true,
+		BinaryIds:     binaryIDsToProto(binaryIDs),
+	})
+	if err != nil {
+		return nil, err
+	}
+	data := make([]BinaryData, len(resp.Data))
+	for i, protoData := range resp.Data {
+		data[i] = binaryDataFromProto(protoData)
+	}
+	return data, nil
+}
+
+// DeleteTabularData deletes tabular data older than a number of days, based on the given organization ID.
```

Comment:
here (and below), can we update the comment to give clarity on what the return value means? I don't think it's intuitive/obvious that we're returning a count of how much data was deleted.